main.js

```diff
@@ -1,82 +1,157 @@
-            document.getElementById('calc').value = document.getElementById('calc').value + number;
+    const currentNumber = document.getElementById('calc').value;
+    // console.log(val1);
+    // If op is undefined then work on creating var1 //
+    if ( lastAnsw !== undefined ) {
+        if ( op !== '' ) {
+            lastAnsw = undefined;
+            document.getElementById('calc').value = number;
+            val2 = document.getElementById('calc').value;
         } else {
+        // do nothing so that user does not override answer from calculation before choosing operator //
+        }
+    } else if ( op == '' ) {
+        if ( currentNumber == 0 ) {
             document.getElementById('calc').value = number;
+            val1 = number;
+        } else {
+            document.getElementById('calc').value = currentNumber + number;
+            val1 = document.getElementById('calc').value;
         }
     } else {
-        document.getElementById('calc').value = document.getElementById('calc').value + number;
+        if ( currentNumber == 0 ) {
+            document.getElementById('calc').value = number;
+            val2 = number;
+        } else {
+            document.getElementById('calc').value = val2 + number;
+            val2 = document.getElementById('calc').value;
+        }
     }
 }
 
-function allClearNum() {
-    // clears input to 0 and clears all variables
+function clearNum() {
     document.getElementById('calc').value = 0;
-    val1 = undefined;
-    val2 = undefined;
-    op = undefined;
+}
+
+function myOperator(myOp) {
+    console.log(history);
+    if ( lastAnsw !== undefined ) {
+        val1 = lastAnsw;
+        lastAnsw = undefined;
+        op = myOp;
+        document.getElementById('op').value = myOp;
+        histroy = history + val1 + op;
+        document.getElementById('hist').innerHTML = history;
+    } else if ( op === '' ) {
+        op = myOp;
+        document.getElementById('op').value = myOp;
+        history = history + val1 + op;
+        document.getElementById('hist').innerHTML = history;
+    } else if ( val2 === '' ) {
+        op = myOp;
+        document.getElementById('op').value = myOp;
+        history = val1 + op;
+        document.getElementById('hist').innerHTML = history;
+    } else {
+        calculate();
+        document.getElementById('op').value = myOp;
+        op = myOp;
+        history = history + op;
+        document.getElementById('hist').innerHTML = history;
+    }
+}
+
+function calculate() {
+    history = history + val2;
+    document.getElementById('hist').innerHTML = history;
+    let answer;
+    switch (op) {
+    case '+':
+        answer = Number(val1) + Number(val2);
+        break;
+    case '-':
+        answer = Number(val1) - Number(val2);
+        break;
+    case '*':
+        answer = Number(val1) * Number(val2);
+        break;
+    case '&#247;':
+        answer = Number(val1) / Number(val2);
+        break;
+    }
```

Comment:
We will want to make sure you have a default case here in the switch statement and indent your case statements inside the switch.

Also we will want to rethink how we are doing the operations here in order to preserve order of operations.